ocaml/gencert/gencert.ml

```diff
@@ -76,19 +76,34 @@ let () =
   let program_name = Sys.argv.(0) in
   let dbg = Printf.sprintf "%s - %f" program_name (Unix.gettimeofday ()) in
   (* if necessary use Unix.localtime to debug *)
-  D.debug "%s" dbg ;
-  match Sys.argv with
-  | [|_; path; _; _|] when Sys.file_exists path ->
-      D.info "file already exists at path (%s) - doing nothing" path ;
-      exit 0
-  | [|_; path; cert_gid; sni|] -> (
+  let sni_or_exit sni =
     match SNI.of_string sni with
     | Some sni ->
-        main ~dbg ~path ~cert_gid:(int_of_string cert_gid) ~sni ()
+        sni
     | None ->
         D.error "SNI must be default or xapi:pool, but got '%s'" sni ;
         exit 1
-  )
+  in
+  let gid_or_exit gid =
+    match int_of_string_opt gid with
+    | Some gid ->
+        gid
+    | None ->
```

Comment:
You want to catch exceptions here, too - in case this is an illegal string. Are negative numbers permitted? Might want to catch those, too.

Reply:
I don't think this function can raise any exceptions: https://github.com/ocaml/ocaml/blob/f2ff4a43d9a76ae84fd9fa66ba8440d88c90d766/stdlib/stdlib.ml#L270-L273

Reply:
Looking at the C implementation I believe Failure is the only kind of exception that int_of_string can return, so the `_opt` variant should not raise any.